controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
 	- Watch for new types being provided by stacks
 	- When there is a new type provided, watch for instances being created
 	- When an instance is created, feed it into the renderer
+
+	TODO This could use a bit of refactoring so it flows more nicely
 	*/
 
 	// configuration is a typed object
-	configuration, err := r.getStackConfiguration()
+	configuration, err := r.getStackConfiguration(ctx, claim)
+	// TODO check for errors
+
+	targetResourceBehavior, err := r.getBehavior(ctx, claim, configuration)
+
+	if targetResourceBehavior == nil {
+		// TODO error condition with a real error returned
+		r.Log.V(0).Info("Couldn't find a configured behavior!",
+			"claim", claim,
+			"configuration", configuration,
+		)
+		return err
+	}
+
+	engineConfig, err := r.createBehaviorEngineConfiguration(ctx, claim, configuration)
+
+	if err != nil {
+		r.Log.Error(err, "Error creating engine configuration!", "claim", claim)
+		return err
+	}
+
+	targetStackImage := configuration.Spec.Source.Image
 
-	// TODO the status will have more than *just* the result in it,
-	// but we can start with just the result
-	result, err := r.processConfiguration(ctx, claim, configuration)
+	result, err := r.executeBehavior(ctx, claim, engineConfig, targetStackImage, targetResourceBehavior)
+	// TODO check for errors
 
 	err = r.setClaimStatus(claim, result)
 
 	return err
 }
 
-func (r *HelmChartInstallReconciler) getClaim() (unstructured.Unstructured, error) {
-	// The claim is the CR that triggered this whole thing.
-	return unstructured.Unstructured{}, nil
-}
-
-func (r *HelmChartInstallReconciler) getStackConfiguration() (unstructured.Unstructured, error) {
+func (r *HelmChartInstallReconciler) getStackConfiguration(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+) (*v1alpha1.StackConfiguration, error) {
```

Comment:
I feel that this line would be more readable if it were not wrapped. Go has no line length concerns, so I typically:

* Don't wrap function signatures until they're pushing 120-140 chars.
* Think really hard about whether a function signature that is over 120-140 chars is a code smell, perhaps due to overly descriptive argument names, or too many arguments.